### PR TITLE
caprevoke: Fix missing early-clobber in vm_do_cheri_revoke

### LINKED ISSUE #2227: vm_do_cheri_revoke inline ASM missing early-clobber
It seems to me that `stxr_status` has to be marked as early-clobbered (=&r instead of =r). I am running GENERIC-MORELLO-PURECAP with KASAN and I found out that, due to ASAN instrumentation, the compiler will try to assign the same register to `cutp` and `stxr_status`, therefore causing a fault when loading from `cutp`. This doesn’t happen when compiling without KASAN, probably because the compiler didn’t have to make this register allocation optimisation.

This is the relevant inline asm:
```c
		__asm__ __volatile__ (
		        "mov %w[stxr_status], #1\n\t"
#ifndef __CHERI_PURE_CAPABILITY__
			"bx #4\n\t"
			".arch_extension c64\n\t"
#endif
			"0: ldxr %[cscratch], [%[cutp]]\n\t"
			"cmp %[cscratch], %[cut]\n\t"
			"bne 1f\n\t"
			"stxr %w[stxr_status], %[cutr], [%[cutp]]\n\t"
			"cbnz %w[stxr_status], 0b\n\t"
			"1:\n\t"
#ifndef __CHERI_PURE_CAPABILITY__
			"bx #4\n\t"
			".arch_extension noc64\n\t"
			".arch_extension a64c\n\t"
#endif
		  : [stxr_status] "=r" (stxr_status),
		    [cscratch] "=&C" (cscratch), [cutr] "+C" (cutr)
		  : [cut] "C" (cut), [cutp] "C" (cutp)
		  : "memory");
```
https://github.com/CTSRD-CHERI/cheribsd/blob/b2ad856aac657e26a9117ea6662289575e6eb514/sys/arm64/arm64/cheri_revoke_machdep.c#L139

### PR DESCRIPTION
**Pull request by RoundofThree**
Fixes #2227.

### CODE REVIEW

**Review by nwf — APPROVED** [review]
LGTM

File: sys/riscv/riscv/cheri_revoke_machdep.c [hunk 1]
```diff
@@ -141,7 +141,7 @@ vm_do_cheri_revoke(int *res, const struct vm_cheri_revoke_cookie *crc,
 			"beq x0, %[ok], 1f\n\t"
 			"sc.c.cap %[cutr], (%[cutp])\n\t"
 			"1:\n\t"
-		  : [ok] "=r" (ok), [cscratch] "=&C" (cscratch),
+		  : [ok] "=&r" (ok), [cscratch] "=&C" (cscratch),
```
Comment (markjdb) [inline sys/riscv/riscv/cheri_revoke_machdep.c:144]: Shouldn't this be `+&r` as well?

Comment (jrtc27) [reply sys/riscv/riscv/cheri_revoke_machdep.c:144]: No, the value prior to the asm statement isn’t read (and isn’t initialised)

Comment (jrtc27) [reply sys/riscv/riscv/cheri_revoke_machdep.c:144]: The difference is Morello can use condition codes to compare whereas RISC-V has to use a CSEQX to get the bool

Comment (markjdb) [reply sys/riscv/riscv/cheri_revoke_machdep.c:144]: I see now, thank you.

**Review by markjdb — APPROVED** [review]

**Pull request merged**